src/component.cpp

```diff
@@ -177,8 +195,8 @@ bool Component::addVariable(const VariablePtr &variable)
         return false;
     }
 
-    // Prevent adding multiple times to list.
-    if (hasVariable(variable)) {
+    // Prevent adding the same variable.
+    if (std::find(mPimpl->mVariables.begin(), mPimpl->mVariables.end(), variable) != mPimpl->mVariables.end()) {
         return false;
     }
```

Comment:
I don't quite understand this?  Why would you even prevent the addition of duplicates any more, since two pointers to the same content are apparently interchangeable?

Reply:
10a2411